doc/admin-guide/plugins/header_rewrite.en.rst

```diff
@@ -806,13 +806,39 @@ QSA    Append the results of the rule to the query string.
 
 Values and Variable Expansion
 -----------------------------
-You can concatenate values using strings, condition values and variable expansions via the ``+`` operator. Variables (eg %<tag>) in the concatenation must be enclosed in double quotes ``"``::
+
+.. note::
+
+   This feature is replaced with a new string concatenations as of ATS v8.1.0. In v9.0.0  the special
+   %<> string expansions below are no longer available, instead use the following mapping:
+
+======================= ==================================================================================
+Variable                New condition variable to use
+======================= ==================================================================================
+%<proto>                %{CLIENT-URL:SCHEME}
+%<port>                 %{CLIENT-URL:PORT}
+%<chi>                  %{IP:CLIENT}, %{INBOUND:REMOTE-ADDR} or e.g. %{CIDR:24,48}
+%<cqhl>                 %{CLIENT-HEADER:Content-Length}
+%<cqhm>                 %{METHOD}
+%<cque>                 %[CLIENT-URL}
+%<cquup>                %{CLIENT-URL:PATH}
+
+Remember that the %<INBOUND:...> tags can now simply be replaced with the %{INBOUND:...} equivalent.
+
+You can concatenate values using strings, condition values and variable expansions via the ``+`` operator.
+Variables (eg %<tag>) in the concatenation must be enclosed in double quotes ``"``::
 
     add-header CustomHeader "Hello from " + %{IP:SERVER} + ":" + "%<INBOUND:LOCAL-PORT>"
 
+However, the above example is somewhat contrived to show the old tags, it should instead be written as
+
+    add-header CustomHeader "Hello from " + %{IP:SERVER} + ":" + %{INBOUND:LOCAL-PORT}
+
+
 Concatenation is not supported in condition testing.
 
-Supported substitutions are currently:
+Supported substitutions are currently the following table, however they are deprecated and you should
+instead use the equivalent %{} conditions as shown above:
 
 ======================= ==================================================================================
 Variable                Description
```

Comment:
If all these are Deprecated, perhaps we should be more obvious than the sentence above. for instance 

```
 ======================= ============================================================================================= %<proto>                Protocol
 %<port>                 (Deprecated) Port
 %<chi>                  (Deprecated) Client IP
 %<cqhl>                 (Deprecated) Client request length
 %<cqhm>                 (Deprecated) Client HTTP method
 %<cque>                 (Deprecated) Client effective URI
 %<cquup>                (Deprecated) Client unmapped URI path
 %<INBOUND:LOCAL-ADDR>   (Deprecated) The local (ATS) address for the inbound connection.
 %<INBOUND:LOCAL-PORT>   (Deprecated) The local (ATS) port for the inbound connection.
 %<INBOUND:REMOTE-ADDR>  (Deprecated) The client address for the inbound connectoin.
 %<INBOUND:REMOTE-PORT>  (Deprecated) The client port for the inbound connectoin.
 %<INBOUND:TLS>          (Deprecated) The TLS protocol for the inbound connection if it is over TLS, otherwise the
                         empty string.
 %<INBOUND:H2>           (Deprecated) The string "h2" if the inbound connection is HTTP/2, otherwise the empty string.
 %<INBOUND:IPV4>         (Deprecated) The string "ipv4" if the inbound connection is IPv4, otherwise the emtpy string.
 %<INBOUND:IPV6>         (Deprecated) The string "ipv6" if the inbound connection is IPv6, otherwise the empty string.
 %<INBOUND:IP-FAMILY>    (Deprecated) The IP family of the inbound connection (either "ipv4" or "ipv6").
 %<INBOUND:STACK>        (Deprecated) The full protocol stack of the inbound connection separated by ','.
 ======================= =============================================================================================
```

Reply:
Updated